platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/event/tipset/TipsetEventCreatorTestUtils.java

```diff
@@ -0,0 +1,266 @@
+// SPDX-License-Identifier: Apache-2.0
+package com.swirlds.platform.event.tipset;
+
+import static com.swirlds.common.test.fixtures.crypto.CryptoRandomUtils.randomSignature;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hiero.base.CompareTo.isGreaterThanOrEqualTo;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.hedera.hapi.node.base.SemanticVersion;
+import com.hedera.hapi.node.state.roster.Roster;
+import com.hedera.hapi.node.state.roster.RosterEntry;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.stream.HashSigner;
+import com.swirlds.common.test.fixtures.platform.TestPlatformContextBuilder;
+import com.swirlds.platform.event.creation.tipset.ChildlessEventTracker;
+import com.swirlds.platform.event.creation.tipset.TipsetEventCreator;
+import com.swirlds.platform.event.creation.tipset.TipsetTracker;
+import com.swirlds.platform.event.creation.tipset.TipsetWeightCalculator;
+import com.swirlds.platform.event.orphan.DefaultOrphanBuffer;
+import com.swirlds.platform.event.orphan.OrphanBuffer;
+import com.swirlds.platform.gossip.IntakeEventCounter;
+import com.swirlds.platform.test.fixtures.event.TestingEventBuilder;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Random;
+import java.util.stream.IntStream;
+import org.hiero.consensus.event.creator.impl.EventCreator;
+import org.hiero.consensus.event.creator.impl.TransactionSupplier;
+import org.hiero.consensus.model.event.AncientMode;
+import org.hiero.consensus.model.event.EventConstants;
+import org.hiero.consensus.model.event.EventDescriptorWrapper;
+import org.hiero.consensus.model.event.PlatformEvent;
+import org.hiero.consensus.model.node.NodeId;
+import org.hiero.consensus.model.transaction.TransactionWrapper;
+import org.junit.jupiter.api.Assertions;
+
+public class TipsetEventCreatorTestUtils {
```

Comment:
Are these utility methods only useful in the context of the `TipsetEventCreatorTest`? Or would it make sense to move them to testFixtures and make them generally available? In particular, the methods to create transactions and events could be useful, as well as `validateNewEvent()`.